File: cpp/PE021.cpp
#include <iostream>
using namespace std;

int d(int n)
{
    int result = 0;
    int i;
    for(i = 1; i < n; i++)
        if(n%i == 0)
            result += i;
    return result;
}

int main()
{
    int i, sum = 0, b;
    for(i = 1; i < 10000; i++){
        if((b = d(i)) > i && d(b) == i)
            sum += i + b;
    }
    cout << sum << endl;
    return 0;
}
            

File: java/PE021.java
public class PE021
{
    public static void main(String[] args)
    {
        int sum = 0;
        for(int i = 2; i < 10000; i++) {
            int n = d(i);
            if(n > i && d(n) == i) {
                sum += n + i;
            }
        }
        System.out.println(sum);
    }
    private static int d(int a)
    {
        int result= 0;
        for(int i = 1; i < a; i++) {
            if(a%i == 0)
                result += i;
        }
        return result;
    }
}
